SWI-cpp2.h

```diff
@@ -1778,14 +1781,13 @@ class PlBlobV
     // order of types; but use cast_check() anyway (which will be
     // optimised away if NDEBUG).
     bool rc_try;
-    int rc = -1; // Stop the compiler warning about uninitialized
+    int rc;
```

Comment:
g++-11 still complains on 

```
/home/janw/src/swipl-devel/packages/cpp/SWI-cpp2.h: In static member function ‘static int PlBlobV<C_t>::compare(atom_t, atom_t) [with C_t = MapStrStr]’:
/home/janw/src/swipl-devel/packages/cpp/SWI-cpp2.h:1784:9: warning: ‘rc’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 1784 |     int rc;
      |         ^~
```

Not sure it is right or not.   Why do we need a try..catch here?   Getting the data from a blob can never fail (may be NULL).

Reply:
This is user code: `a_data->compare_fields(b_data)` and is allowed to throw an exception. (Similar with the other try-catch in `write()`, etc.)
There's no overhead in the `try`, only in the `catch` (I've documented this with `PlFail()`)
I'll fix the compiler warning -- there's one small other change I want to make.